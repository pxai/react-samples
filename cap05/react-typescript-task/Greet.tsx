import * as React from 'react';

interface GreetState {
    message: string
}

class Greet extends React.Component<any, GreetState> {
    public constructor(props) {
        super(props);
        this.state = {
            message: "This is a greet"
        };
    }

    private doSomething () {
        this.setState({message: "I was clicked"});
    }

    public render (): React.ReactNode {
        return <div className="Greet" onClick={() => this.doSomething()}>
		    {this.state.message}
        </div>;
    }
}

export default Greet;