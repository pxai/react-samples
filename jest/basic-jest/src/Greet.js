import React, { Component } from 'react';
import './Greet.css';

class Greet extends Component {
  render() {
    return (
      <div className="Greet">
		This is a greet
      </div>
    );
  }
}

export default Greet;
