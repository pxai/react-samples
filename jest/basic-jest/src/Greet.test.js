import React from 'react';
import ReactDOM from 'react-dom';
import Greet from './Greet';
import renderer from 'react-test-renderer';

test('renders Greet without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Greet />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('it matches snapshot', () => {
	const component = renderer.create(<Greet />);
 let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('it changes state', () => {
	const component = renderer.create(<Greet />);
 let tree = component.toJSON();
 tree.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
