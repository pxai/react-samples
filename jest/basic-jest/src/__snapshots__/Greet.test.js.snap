// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<div
  className="Greet"
>
  This is a greet
</div>
`;
